nativelib/src/main/scala/scala/scalanative/unsafe/Tag.scala.gyb

```diff
@@ -24,45 +24,69 @@ sealed abstract class Tag[T] {
   @noinline def store(ptr: unsafe.Ptr[T], value: T): Unit = throwUndefined()
 }
 
+// Separate object to avoid dependency cycles
```

Comment:
This is needed not for the name clash, but because of a circular dependency at runtime between the initialization logic `Tag` and `USize`. By splitting them apart, we avoid a segfault during initialization.